### PR TITLE
Built-in submit and stop buttons in `gr.ChatInterface(multimodal=False)`, adding `submit_btn` and `stop_btn` props to `gr.Textbox()` and `gr.MultimodalText()`

### PR DESCRIPTION
**Pull request by whitphx**
## Description

Closes: #9219

For each consideration stated in #9219, the answer follows:

> * Should the submit icon be enabled by default? (I'm thinking no but not sure)

I agree with it, and have done in this PR:
https://github.com/gradio-app/gradio/pull/9235/files#diff-67385871d553144fb7c5ea0fd0aca8b52b3a8575cb15a13715661eb8264c657aR63
https://github.com/gradio-app/gradio/pull/9235/files#diff-2faed51318325400c0a721475c81802877e5cc1478791e564edce5a6f28ba340R84

> * "Even cleaner would be if the submit icon turns into the stop button natively"

~~It's not done in this PR at this point.~~ Done in this PR.

> * Can we remove the box shadow in gr.Textbox and instead add some padding, similar to gr.MultimodalTextbox (we should make sure it looks good in a variety of demos -- it certainly looks better in gr.ChatInterface!)

Now the non-shadowed style is applied when `submit_btn` is falthy. Do you think we should provide an option dedicated for it?
https://github.com/gradio-app/gradio/pull/9235/files#diff-4eee3cd89ad94bd42114ffff451659c388f43eb5f61e742214ae19565170066bR39

### Bare usage of `gr.Textbox`
```python
demo = gr.Interface(
    fn=lambda x: x,
    inputs=gr.Textbox(),
    outputs=gr.Textbox(),
)
```
CleanShot 2024-09-01 at 20 37 49@2x

### Bare usage of `gr.MultimodalTextbox`
```python
demo = gr.Interface(
    fn=lambda x: x,
    inputs=gr.MultimodalTextbox(),
    outputs=gr.MultimodalTextbox(),
)
```

CleanShot 2024-09-01 at 20 38 23@2x

### CODE REVIEW

**Comment by whitphx** [comment]
At this point, the stop button of `gr.ChatInterface` has been kept as-is, not being moved into the textbox components.
The stop button became available with `multimodal=True`.

CleanShot 2024-09-01 at 20 42 39@2x

CleanShot 2024-09-01 at 20 42 00@2x

**Comment by whitphx** [comment]
I added the built-in stop button.

CleanShot 2024-09-01 at 21 56 00@2x

CleanShot 2024-09-01 at 21 55 25@2x

**Comment by whitphx** [comment]
Will modify the tests too.

**Comment by abidlabs** [comment]
Amazing, thank you @whitphx for this nice PR! Just one nit: can we change the pause icon to the more canonical stop icon: 

image

Will review the rest of the PR soon. cc @freddyaboulton @dawoodkhan82 as well

**Comment by whitphx** [comment]
@abidlabs Replaced the Pause icon with the Square 👍

File: gradio/components/textbox.py [hunk 1]
```diff
@@ -88,6 +91,7 @@ def __init__(
             show_copy_button: If True, includes a copy button to copy the text in the textbox. Only applies if show_label is True.
             autoscroll: If True, will automatically scroll to the bottom of the textbox when the value changes, unless the user scrolls up. If False, will not scroll to the bottom of the textbox when the value changes.
             max_length: maximum number of characters (including newlines) allowed in the textbox. If None, there is no maximum length.
+            submit_btn: If False, will not show a submit button. If a string, will use that string as the submit button text.
```
Comment (abidlabs) [inline gradio/components/textbox.py:94]: We should also mention that it removes the input field border:

image

Comment (abidlabs) [reply gradio/components/textbox.py:94]: i.e. compared to:

image

**Review by abidlabs — APPROVED** [review]
Great PR, thanks @whitphx! Code changes look good & behaves exactly like I expected in the default case:

https://github.com/user-attachments/assets/cf501155-704d-4c8f-979c-b64c1d9b86b5

If @dawoodkhan82 has a chance to review soon, that'd be great, otherwise I can do some more testing for edge cases before merging this in

File: gradio/components/textbox.py (hunk 1 above)
Comment (whitphx) [reply gradio/components/textbox.py]: Done!

**Comment by abidlabs** [comment]
Hi @whitphx I tested some more, and there's a small issue that happens when a custom textbox is provided & streaming is enabled for the chatbot. You can see it here:

```
import gradio as gr

def slow_echo(x, y):
    for i in range(10):
        yield "a"

gr.ChatInterface(slow_echo, textbox=gr.Textbox()).launch()
```

https://github.com/user-attachments/assets/4a2d5759-ccb1-4969-bc25-be1b9e5c5f23

In this case, the textbox looks like a regular textbox (which is fine), but a stop button appears once the stream starts and is replaced by the submit button when the stream ends. The same happens for the multimodal case:

```py
import gradio as gr

def slow_echo(x, y):
    for i in range(10):
        yield {"text": "a"}

gr.ChatInterface(slow_echo, textbox=gr.MultimodalTextbox(), multimodal=True).launch()
```

So basically, we should only make the stop button appear if submit_btn was true from the beginning of the textbox. Otherwise, we should not show the stop button during streaming (nor the submit button after the stream ends)

**Comment by whitphx** [comment]
Pushed the fix.

But now I think using the user-specified `submit_btn` and `stop_btn` values provides more customizability to the users,

For example, the code below should show the chat interface without a submit button but show the stop button with the custom text during the generator execution.
```python
gr.ChatInterface(
    fn,
    textbox=gr.MultimodalTextbox(submit_btn=False, stop_btn="stop text"),
    multimodal=True
)
```

Another example is below, where it should show the chat interface with a submit button with the custom text and show the stop button with the custom text during the generator execution.
```python
gr.ChatInterface(
    fn,
    textbox=gr.MultimodalTextbox(submit_btn="submit text", stop_btn="stop text"),
    multimodal=True
)
```

Will give it a try.

**Comment by whitphx** [comment]
Also I reverted `submit_btn` and `stop_btn` of `gr.ChatInterface` but with different signatures that allows customization like this:
```python
demo = gr.ChatInterface(
    slow_echo,
    submit_btn="Submit",
    stop_btn="Stop",
)
```

wdyt about this design?

File: gradio/chat_interface.py [hunk 2]
```diff
@@ -104,14 +104,14 @@ def __init__(
...
             fill_height: if True, the chat interface will expand to the height of window.
             delete_cache: a tuple corresponding [frequency, age] both expressed in number of seconds. Every `frequency` seconds, the temporary files created by this Blocks instance will be deleted if more than `age` seconds have passed since the file was created. For example, setting this to (86400, 86400) will delete temporary files every day. The cache will be deleted entirely when the server restarts. If None, no cache deletion will occur.
             show_progress: how to show the progress animation while event is running: "full" shows a spinner which covers the output component area as well as a runtime display in the upper right corner, "minimal" only shows the runtime display, "hidden" shows no progress animation at all
             fill_width: Whether to horizontally expand to fill container fully. If False, centers and constrains app to a maximum width.
+            submit_btn: If True, will show a submit button with an icon. If a string, will use that string as the submit button text. If False, will not show a submit button.
```
Comment (abidlabs) [inline gradio/chat_interface.py:113]:
```suggestion
            submit_btn: If True, will show a submit button with a submit icon within the textbox. If a string, will use that string as the submit button text in place of the icon. If False, will not show a submit button.
```

File: gradio/chat_interface.py [hunk 3]
```diff
@@ -104,14 +104,14 @@ def __init__(
...
+            stop_btn: If True, will show a stop button with an icon during generator executions. If a string, will use that string as the stop button text. If False, will not show a stop button.
```
Comment (abidlabs) [inline gradio/chat_interface.py:114]:
```suggestion
            stop_btn: If True, will show a button with a stop icon during generator executions, to stop generating. If a string, will use that string as the submit button text in place of the stop icon. If False, will not show a stop button.
```

File: gradio/events.py [hunk 4]
```diff
@@ -711,6 +711,10 @@ class Events:
         "submit",
         doc="This listener is triggered when the user presses the Enter key while the {{ component }} is focused.",
     )
+    stop = EventListener(
+        "stop",
+        doc="This listener is triggered when the user requests stopping the running task.",
```
Comment (abidlabs) [inline gradio/events.py:716]:
```suggestion
        doc="This listener is triggered when the user clicks on the stop button or icon.",
```

**Comment by abidlabs** [comment]
Everything is working great @whitphx. LGTM!

> wdyt about this design?

No strong opinion on my side. On one hand, it seems like an unnecessary argument if a developer can already set it via the `textbox` parameter. But on the other hand, it means we have to deprecate one fewer argument.

**Review by abidlabs — APPROVED** [review]

**Comment by whitphx** [comment]
Thank you!
And I also don't have a clear answer to that tradeoff.
One thing I thought was, providing `submit_btn` and `stop_btn` arguments from `ChatInterface()` makes the use of the API a bit simpler, not being required to define the textbox instance by hand, so that the user don't have to choose `Textbox` or `MultimodalTextbox` aligning with the `multimodal` option manually.
